src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -57,45 +66,90 @@ private Job runJob(Job queryJob) {
   private enum Source {
     GDC,
     PDC;
-
-    final String match;
-
-    Source() {
-      match = String.format("\"system\": \"%s\"", name());
-    }
-
-    boolean match(String s) {
-      return s.contains(match);
-    }
   }
 
   private static class Results {
-    final List<String> jsonData = new ArrayList<>();
+    final List<JsonNode> jsonData = new ArrayList<>();
     /** For each system, the number of rows in jsonData that have data from that system. */
     final Map<Source, Integer> resultsCount = new EnumMap<>(Source.class);
+
+     /** Traverse the json data and collect the number of systems data present in resultsCount. */
+    private void generateUsageData() {
+      // Each node is a single row in the result.
+      for (JsonNode jsonNode : jsonData) {
+        List<String> systems = jsonNode.findValuesAsText("system");
+        Arrays.stream(Source.values())
```

Comment:
Are you worried about the case where `generateUsageData` gets called multiple times?  That could increase the counts and I just want to make sure that's expected.  Do you need to have a `resultsCount.clear()` in the beginning?  And if this is called from an endpoint, you probably need this in a `synchronized` since I don't think this is threadsafe

Reply:
The usage data is only used for logging the current request and is allocated once per every request, so there's no thread issue here. The various refactorings have made this less clear, so I'm going to get rid of the `Results` object and the lifetime of the usage should be more apparent.